Nos/Service/ReportPublisher.swift

```diff
@@ -0,0 +1,130 @@
+    static var publicKey = PublicKey(npub: "npub14h23jzlyvumks4rvrz6ktk36dxfyru8qdf679k7q8uvxv0gm0vnsyqe2sh")!
+}
+
+class ReportPublisher {
+    @Dependency(\.relayService) private var relayService
+    @Dependency(\.currentUser) private var currentUser
+    @Dependency(\.analytics) private var analytics
+    
+    /// Publishes a report for the given target and category.
+    func publishPublicReport(for target: ReportTarget, category: ReportCategory, context: NSManagedObjectContext) {
+        guard let keyPair = currentUser.keyPair else {
+            Log.error("Cannot publish report - No signed in user")
+            return
+        }
+        
+        let publicReport = createPublicReport(for: target, category: category, pubKey: keyPair.publicKeyHex)
+        
+        Task {
+            do {
+                try await relayService.publishToAll(event: publicReport, signingKey: keyPair, context: context)
+                analytics.reported(target)
+            } catch {
+                Log.error("Failed to publish report: \(error.localizedDescription)")
+            }
+        }
+    }
+    
+    func publishPrivateReport(for target: ReportTarget, category: ReportCategory, context: NSManagedObjectContext) {
+        switch target {
+        case .note(let note):
+            guard let keyPair = currentUser.keyPair else {
+                Log.error("Cannot publish report - No signed in user")
+                return
+            }
+            
+            guard let reportRequestDM = createReportRequestDM(note: note, category: category, keyPair: keyPair) else {
+                Log.error("Failed to create gift wrapped report request")
+                return
+            }
+            
+            Task {
+                do {
+                    try await relayService.publish(
+                        event: reportRequestDM,
+                        to: Relay.nosAddress,
+                        context: context
+                    )
+                    analytics.reported(target)
+                } catch {
+                    Log.error("Failed to publish report: \(error.localizedDescription)")
+                }
+            }
+        case .author:
+            Log.error("Private reports for people are no implemented yet")
+            return
+        }
+    }
+    
+    // MARK: Helpers
+    
+    internal struct ReportRequest {
+        let reportedEvent: JSONEvent
+        let reporterPubkey: RawAuthorID
+        let reporterText: String
+        
+        func toJSON() throws -> String {
+            let dictionary: [String: Any] = [
+                "reportedEvent": reportedEvent.dictionary,
+                "reporterPubkey": reporterPubkey,
+                "reporterText": reporterText
+            ]
+            
+            let data = try JSONSerialization.data(withJSONObject: dictionary)
+            return String(data: data, encoding: .utf8)!
+        }
+    }
+    
+    internal func createPublicReport(
```

Comment:
We don't need `internal` in this and the other functions in this file. At least until we start making swift packages again. Can you remove them?

Reply:
(`internal` is the default access modifier, so everything is `internal` unless you specify otherwise)

Reply:
Can I still access this function from the ReportPublisherTests with a private func? I think I added it just to ease testing

Reply:
got it, probably must have been one of those weird compilation errors that I though was a visibility problem and just coincidentally worked